pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -2367,6 +2376,46 @@ func TestListPlugins(t *testing.T) {
 	}
 }
 
+func TestPluginWeights(t *testing.T) {
+	tests := []struct {
+		name    string
+		plugins *config.Plugins
+	}{
+		{
+			name: "Add multiple plugins with weights",
+			plugins: &config.Plugins{
+				QueueSort: config.PluginSet{Enabled: []config.Plugin{{Name: queueSortPlugin}}},
+				Score:     config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin, Weight: 3}}},
+				PostBind:  config.PluginSet{Enabled: []config.Plugin{{Name: testPlugin, Weight: 6}}},
+				Bind:      config.PluginSet{Enabled: []config.Plugin{{Name: bindPlugin}}},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			profile := config.KubeSchedulerProfile{Plugins: tt.plugins}
+			f, err := NewFramework(registry, &profile)
+			if err != nil {
+				t.Fatalf("Failed to create framework for testing: %v", err)
+			}
+			plugins := f.ListPlugins()
+
+			compareScoreWeight(t, plugins["ScorePlugin"], tt.plugins.Score.Enabled)
```

Comment:
```suggestion
			compareScoreWeight(t, plugins["ScorePlugin"][0], tt.plugins.Score.Enabled[0])
```

You just have one score plugin, this will help to drop the `for ... for` loop below.